src/controller/tests/data_model/TestRead.cpp

```diff
@@ -2082,21 +2074,52 @@ void TestReadInteraction::TestReadHandler_KillOverQuotaSubscriptions(nlTestSuite
 
     app::InteractionModelEngine::GetInstance()->RegisterReadHandlerAppCallback(&gTestReadInteraction);
 
+    // Here, we set up two background infinity read requests to simulate parallel Read + Subscriptions.
+    // We don't care about the data read, we only care about the existence of such read transactions.
     TestReadCallback readCallback;
     TestReadCallback readCallbackFabric2;
+    TestPerpetualListReadCallback infinityReadCallback;
     std::vector<std::unique_ptr<app::ReadClient>> readClients;
 
+    EstablishReadOrSubscriptions(apSuite, ctx.GetSessionAliceToBob(), 1, 1,
+                                 app::AttributePathParams(kTestEndpointId, TestCluster::Id, kNeverEndAttributeid),
+                                 app::ReadClient::InteractionType::Read, &infinityReadCallback, readClients);
+    EstablishReadOrSubscriptions(apSuite, ctx.GetSessionBobToAlice(), 1, 1,
+                                 app::AttributePathParams(kTestEndpointId, TestCluster::Id, kNeverEndAttributeid),
+                                 app::ReadClient::InteractionType::Read, &infinityReadCallback, readClients);
+    ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(5), [&]() {
+        return app::InteractionModelEngine::GetInstance()->GetNumActiveReadHandlers(app::ReadHandler::InteractionType::Read) == 2;
+    });
+    // Ensure our read transactions are established.
+    NL_TEST_ASSERT(apSuite,
+                   app::InteractionModelEngine::GetInstance()->GetNumActiveReadHandlers(app::ReadHandler::InteractionType::Read) ==
+                       2);
+
     // Intentially establish subscriptions using exceeded resources.
     app::InteractionModelEngine::GetInstance()->SetForceHandlerQuota(false);
-    EstablishSubscriptions(apSuite, ctx.GetSessionBobToAlice(), 1,
-                           app::InteractionModelEngine::kMinSupportedPathsPerSubscription + 1, &readCallback, readClients);
-    EstablishSubscriptions(apSuite, ctx.GetSessionBobToAlice(), kExpectedParallelSubs,
-                           app::InteractionModelEngine::kMinSupportedPathsPerSubscription, &readCallback, readClients);
+    // We established kExpectedParallelSubs + 1 subsctions, with (kExpectedParallelSubs + 1) * kMinSupportedPathsPerSubscription + 1
+    // interested paths. This exceeds the limit we set below, and the subscription with kMinSupportedPathsPerSubscription + 1
+    // interested path will be evicted firstly.
+    // We intentially use exceeded resource here to make testing "used exactly all resources" easier.
```

Comment:
```suggestion
    //
    // We establish 1 subscription that exceeds the minimum supported paths (but is still established since the 
    // target has sufficient resources), and kExpectedParallelSubs subscriptions that conform to the minimum
    // supported paths. This sets the stage to make it possible to test eviction of subscriptions that are in violation
    // of the minimum later below.
    //
```